lineapy/graph_reader/session_artifacts.py

```diff
@@ -50,9 +56,16 @@ def __init__(
         self.artifact_nodecollections = []
         self.node_context = OrderedDict()
         self.input_parameters = input_parameters
+        self.use_cache = dict()
+        for cache_art in use_cache:
+            if isinstance(cache_art, str):
+                self.use_cache[cache_art] = None
+            else:
+                self.use_cache[cache_art[0]] = int(cache_art[1])
 
         self._update_node_context()
```

Comment:
post our discussion, I think it would be _very_ helpful to comment what each of these 3 functions do inline here. This is the core logic of session_artifacts so let's make sure its clear what the flow is.

Reply:
_update_node_context
_slice_session_artifacts
_update_nodecollection_dependencies 

to be explicit